src/frequenz/sdk/timeseries/_base_types.py

```diff
@@ -150,3 +152,37 @@ class Bounds(Generic[_T]):
 
     upper: _T
     """Upper bound."""
+
+
+@dataclass(frozen=True)
```

Comment:
I would use `kw_only=True` here, as we have 2 attributes with `Bounds[Power]` type, that could be easily mixed up.